ecs-agent/tmds/handlers/fault/v1/types/types.go

```diff
@@ -80,6 +80,34 @@ func (response NetworkFaultInjectionResponse) ToString() string {
 	return string(data)
 }
 
+type NetworkLatencyRequest struct {
+	DelayMilliseconds  *uint64   `json:"DelayMilliseconds"`
+	JitterMilliseconds *uint64   `json:"JitterMilliseconds"`
+	Sources            []*string `json:"Sources"`
+}
+
+func (request NetworkLatencyRequest) ValidateRequest() error {
+	if request.DelayMilliseconds == nil {
+		return fmt.Errorf(missingRequiredFieldError, "DelayMilliseconds")
+	}
+	if request.JitterMilliseconds == nil {
+		return fmt.Errorf(missingRequiredFieldError, "JitterMilliseconds")
+	}
+	if request.Sources == nil || len(request.Sources) == 0 {
+		return fmt.Errorf(missingRequiredFieldError, "Sources")
+	}
```

Comment:
Is `Sources` a required field? What if someone wants to block all sources?

Reply:
Yes this is required. If customers would like to block all, `0.0.0.0/0` would be passed from the client.

Reply:
Added comment for `sources`